cmd/kubeadm/app/checks/checks.go

```diff
@@ -0,0 +1,185 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package checks
+
+import (
+	"fmt"
+	"io"
+	"net"
+	"os"
+	"os/exec"
+)
+
+// PreFlightCheck validates the state of the system to ensure kubeadm will be
+// successful as often as possilble.
+type PreFlightCheck interface {
+	Check() (warnings []string, errors []string)
+}
+
+// ServiceCheck verifies that the given service is enabled and active. If we do not
+// detect a supported init system however, all checks are skipped and a warning is
+// returned.
+type ServiceCheck struct {
+	service string
+}
+
+func (sc ServiceCheck) Check() (warnings []string, errors []string) {
+
+	initSystem := getInitSystem()
+	if initSystem == nil {
+		return []string{"no supported init system detected, skipping service checks"}, nil
+	}
+
+	warnings = []string{}
+
+	if !initSystem.ServiceExists(sc.service) {
```

Comment:
Actually I don't think this check is necessary, since we're blocking on the fact that the service should be active